bitcoin/src/pow.rs

```diff
@@ -234,6 +234,34 @@ impl Target {
     /// [`difficulty`]: Target::difficulty
     #[cfg_attr(all(test, mutate), mutate)]
     pub fn difficulty_float(&self) -> f64 { TARGET_MAX_F64 / self.0.to_f64() }
+
+    /// Computes the minimum valid [`Target`] threshold allowed for a block in which a difficulty
+    /// adjustment occurs. Note that the result is bounded by [`Target::MAX`].
+    ///
+    /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
+    /// adjustment period.
+    pub fn min_difficulty_transition_threshold(&self) -> Self {
```

Comment:
The function could be const I think but you'd have to call `wrapping_shr` instead of using the operator.

Reply:
It became const in Rust 1.32 https://doc.rust-lang.org/std/primitive.u128.html#method.wrapping_shr

Reply:
@tcharding he means *our* `wrapping_shr`.

Right, my PR addressed that, don't bother with it.